pkg/controller/service/controller.go

```diff
@@ -348,15 +336,13 @@ func (s *Controller) syncLoadBalancerIfNeeded(service *v1.Service, key string) (
 		op = ensureLoadBalancer
 		klog.V(2).Infof("Ensuring load balancer for service %s", key)
 		s.eventRecorder.Event(service, v1.EventTypeNormal, "EnsuringLoadBalancer", "Ensuring load balancer")
-		if utilfeature.DefaultFeatureGate.Enabled(serviceLoadBalancerFinalizerFeature) {
-			// Always try to add finalizer prior to load balancer creation.
-			// It will be a no-op if finalizer already exists.
-			// Note this also retrospectively puts on finalizer if the cluster
-			// is upgraded from a version that doesn't attach finalizer to a
-			// version that does.
-			if err := s.addFinalizer(service); err != nil {
-				return op, fmt.Errorf("failed to add load balancer cleanup finalizer: %v", err)
-			}
+		// Always try to add finalizer prior to load balancer creation.
```

Comment:
Wondering if this comment should be updated now. Maybe just state the intent now that it's a default behavior. 

```
// Always add a finalizer prior to creating load balancers, this ensures Services can't be deleted
// until all corresponding load balancer resources are also deleted.
```

Reply:
same for removeFinalizer above

Reply:
Yes that make more sense. Done.